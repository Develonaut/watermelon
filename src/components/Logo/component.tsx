import * as React from 'react';
import logo from 'src/images/watermelon.png';
import './styles.scss';

export const Logo = () => {
  return (
    <div className="logo">
      <div className="logo__container">
        <div className="logo__circle">
          <img src={logo}/>
        </div>
      </div>
    </div>
  );
};
