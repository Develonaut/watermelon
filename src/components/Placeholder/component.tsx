import * as React from 'react';
import './styles.scss';

export interface PublicProps {
    fullWidth?: boolean;
    fullHeight?: boolean;
    hidden?: boolean;
}

type Props = PublicProps;

export class Placeholder extends React.Component<Props> {
  public render() {
    let className = 'placeholder';
    if (this.props.fullHeight) {
        className += ' placeholder--fullHeight';
    }

    if (this.props.fullWidth) {
        className += ' placeholder--fullWidth';
    }

    if (this.props.hidden) {
        className += ' placeholder--hidden';
    }

    return (
      <div className={className} />
    );
  }
}
