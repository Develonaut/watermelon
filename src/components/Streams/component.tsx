import * as React from 'react';
import './styles.scss';

export class Streams extends React.Component {
  public render() {
    if (!this.props.children) {
      return null;
    }

    return (
      <div className="streamers">
        {this.props.children}
      </div>
    );
  }
}
