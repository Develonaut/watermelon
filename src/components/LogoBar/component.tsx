import * as React from 'react';
import { Logo } from 'src/components/Logo';
import './styles.scss';

export const LogoBar = () => {
  return (
    <div className="logo-bar">
      <Logo />
    </div>
  );
};
