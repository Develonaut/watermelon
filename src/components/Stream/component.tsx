import * as React from 'react';
import './styles.scss';

import { Placeholder } from 'src/components/Placeholder';

interface PublicProps {
  streamerName: string;
  focused?: boolean;
}

export interface State {
  loaded: boolean;
}

type Props = PublicProps;

export class Stream extends React.Component<Props, State> {
  public state = {
    loaded: false,
  };

  public render() {
    const frameClassModifier = this.state.loaded ? 'shown' : 'hidden';
    const streamClassModifier = this.props.focused ? 'focused' : 'default';
    return (
      <div className={`stream stream--${streamClassModifier}`}>
        <Placeholder fullWidth fullHeight hidden={this.state.loaded} />
        <div className="stream__container">
          <iframe
              className={`stream__frame stream__frame--${frameClassModifier}`}
              onLoad={this.handleOnLoad}
              src={`//player.twitch.tv/?channel=${this.props.streamerName}&muted=true`}
              height="100%"
              width="100%"
              scrolling="no"
          />
        </div>
      </div>
    );
  }

  private handleOnLoad = () => {
    this.setState({ loaded: true });
  }
}
