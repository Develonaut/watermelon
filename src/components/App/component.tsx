import * as React from 'react';
import { LogoBar } from 'src/components/LogoBar';
import { Stream } from 'src/components/Stream';
import { Streams } from 'src/components/Streams';
import './styles.scss';

export interface Props {
  liveStreams?: string[];
  streams?: string[];
  loading?: boolean;
  error?: boolean;
  fetchLiveStreams: () => void;
  fetchFollowedStreams: () => void;
  fetchFeaturedStreams: () => void;
}

export class AppPresentation extends React.Component<Props> {

  public componentDidMount() {
    this.props.fetchFeaturedStreams();
  }

  public render() {
    if (this.props.loading) {
      return null;
    }

    const featuredStreams = this.props.streams;

    if (featuredStreams && featuredStreams.length) {
      const featuredStreamsNames = (!!featuredStreams && featuredStreams.length && featuredStreams);
      const focusedStreamName = featuredStreams.slice(0, 1)[0];
      return (
        <div className="App">
          <LogoBar />
          <Streams>
            {featuredStreamsNames && featuredStreamsNames.map(
              (name: string) => <Stream streamerName={name} key={name} focused={focusedStreamName === name}/>,
            )}
          </Streams>
        </div>
      );
    }

    return null;
  }
}
