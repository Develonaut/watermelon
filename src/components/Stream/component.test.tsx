import { shallow } from 'enzyme';
import * as React from 'react';
import { Stream } from './component';

describe('Stream --', () => {
    it('renders', () => {
        const wrapper = shallow(<Stream streamerName="irisheffects" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders as the focused stream', () => {
        const wrapper = shallow(<Stream streamerName="irisheffects" focused/>);
        expect(wrapper).toMatchSnapshot();
    });
});
