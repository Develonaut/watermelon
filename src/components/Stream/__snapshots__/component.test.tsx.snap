// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stream -- renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stream
    streamerName="irisheffects"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Placeholder
          fullHeight={true}
          fullWidth={true}
          hidden={false}
        />,
        <div
          className="stream__container"
        >
          <iframe
            className="stream__frame stream__frame--hidden"
            height="100%"
            onLoad={[Function]}
            scrolling="no"
            src="//player.twitch.tv/?channel=irisheffects&muted=true"
            width="100%"
          />
        </div>,
      ],
      "className": "stream stream--default",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fullHeight": true,
          "fullWidth": true,
          "hidden": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <iframe
            className="stream__frame stream__frame--hidden"
            height="100%"
            onLoad={[Function]}
            scrolling="no"
            src="//player.twitch.tv/?channel=irisheffects&muted=true"
            width="100%"
          />,
          "className": "stream__container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stream__frame stream__frame--hidden",
            "height": "100%",
            "onLoad": [Function],
            "scrolling": "no",
            "src": "//player.twitch.tv/?channel=irisheffects&muted=true",
            "width": "100%",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Placeholder
            fullHeight={true}
            fullWidth={true}
            hidden={false}
          />,
          <div
            className="stream__container"
          >
            <iframe
              className="stream__frame stream__frame--hidden"
              height="100%"
              onLoad={[Function]}
              scrolling="no"
              src="//player.twitch.tv/?channel=irisheffects&muted=true"
              width="100%"
            />
          </div>,
        ],
        "className": "stream stream--default",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fullHeight": true,
            "fullWidth": true,
            "hidden": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <iframe
              className="stream__frame stream__frame--hidden"
              height="100%"
              onLoad={[Function]}
              scrolling="no"
              src="//player.twitch.tv/?channel=irisheffects&muted=true"
              width="100%"
            />,
            "className": "stream__container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "stream__frame stream__frame--hidden",
              "height": "100%",
              "onLoad": [Function],
              "scrolling": "no",
              "src": "//player.twitch.tv/?channel=irisheffects&muted=true",
              "width": "100%",
            },
            "ref": null,
            "rendered": null,
            "type": "iframe",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Stream -- renders as the focused stream 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stream
    focused={true}
    streamerName="irisheffects"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Placeholder
          fullHeight={true}
          fullWidth={true}
          hidden={false}
        />,
        <div
          className="stream__container"
        >
          <iframe
            className="stream__frame stream__frame--hidden"
            height="100%"
            onLoad={[Function]}
            scrolling="no"
            src="//player.twitch.tv/?channel=irisheffects&muted=true"
            width="100%"
          />
        </div>,
      ],
      "className": "stream stream--focused",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fullHeight": true,
          "fullWidth": true,
          "hidden": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <iframe
            className="stream__frame stream__frame--hidden"
            height="100%"
            onLoad={[Function]}
            scrolling="no"
            src="//player.twitch.tv/?channel=irisheffects&muted=true"
            width="100%"
          />,
          "className": "stream__container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stream__frame stream__frame--hidden",
            "height": "100%",
            "onLoad": [Function],
            "scrolling": "no",
            "src": "//player.twitch.tv/?channel=irisheffects&muted=true",
            "width": "100%",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Placeholder
            fullHeight={true}
            fullWidth={true}
            hidden={false}
          />,
          <div
            className="stream__container"
          >
            <iframe
              className="stream__frame stream__frame--hidden"
              height="100%"
              onLoad={[Function]}
              scrolling="no"
              src="//player.twitch.tv/?channel=irisheffects&muted=true"
              width="100%"
            />
          </div>,
        ],
        "className": "stream stream--focused",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fullHeight": true,
            "fullWidth": true,
            "hidden": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <iframe
              className="stream__frame stream__frame--hidden"
              height="100%"
              onLoad={[Function]}
              scrolling="no"
              src="//player.twitch.tv/?channel=irisheffects&muted=true"
              width="100%"
            />,
            "className": "stream__container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "stream__frame stream__frame--hidden",
              "height": "100%",
              "onLoad": [Function],
              "scrolling": "no",
              "src": "//player.twitch.tv/?channel=irisheffects&muted=true",
              "width": "100%",
            },
            "ref": null,
            "rendered": null,
            "type": "iframe",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
